pkg/distributor/validator.go

```diff
@@ -161,9 +161,9 @@ func (v Validator) ValidateEntry(ctx context.Context, vCtx validationContext, la
 }
 
 // Validate labels returns an error if the labels are invalid
-func (v Validator) ValidateLabels(ctx validationContext, ls labels.Labels, stream logproto.Stream) error {
+func (v Validator) ValidateLabels(ctx validationContext, ls labels.Labels, stream logproto.Stream, retentionHours string) error {
 	if len(ls) == 0 {
-		validation.DiscardedSamples.WithLabelValues(validation.MissingLabels, ctx.userID).Inc()
+		validation.DiscardedSamples.WithLabelValues(validation.MissingLabels, ctx.userID, retentionHours).Inc()
```

Comment:
I see we only added the retentionHours in the PR but, shouldn't this be `updateMetrics` as well?

Reply:
I agree. We can fix that in a separate PR to involve more people in the discussion.